tests/console/dracut_enhanced.pm

```diff
@@ -0,0 +1,72 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run dracut testsuite
+# - Test suitable only for dracut 0.44 (SLE12SP2+)
+# - Fetches a tarball containing patches to be applied to dracut tests
+# - Install dracut src.rpm and some dependencies
+# - Patch dracut tests (remove references for qemu)
+# - Creates a initramfs and boots the system
+# - Cleanup
+# Based on code from Sergio Lindo Mansilla <slindomansilla@suse.com> and
+# Thomas Blume <tblume@suse.com>
+
+# Maintainer: Ednilson Miura <emiura@suse.com>
+
+use base "consoletest";
+use warnings;
+use strict;
+use testapi;
+use utils 'zypper_call';
+use power_action_utils 'power_action';
+use main_common;
+use version_utils 'is_sle';
+
+sub run {
+    select_console 'root-console';
+    my $dracut_test = get_var('DRACUT_TEST');
+    if (is_sle('<12-SP2')) {
+        die "Unsupported dracut version";
+    }
+    else {
+        assert_script_run 'curl -v -o /tmp/dracut-patches.tar.gz ' . data_url('qam/dracut/dracut-patches.tar.gz');
+        assert_script_run 'tar xvf /tmp/dracut-patches.tar.gz -C /tmp';
+        zypper_call "in rpmbuild dhcp-client strace";
+        zypper_call "si -D dracut";
+        assert_script_run('cd /usr/src/packages/SPECS');
+        assert_script_run('rpmbuild -bp dracut.spec --nodeps');
+        assert_script_run('cd /usr/src/packages/BUILD/dracut-*');
+        assert_script_run('cp /tmp/dracut*.patch .');
+        assert_script_run('for p in *patch; do patch -p0 < $p; done');
+        assert_script_run "cd /usr/src/packages/BUILD/dracut-*/test/$dracut_test";
+        assert_script_run 'mkdir /tmp/logs';
+        assert_script_run "./test.sh --setup 2>&1 | tee /tmp/logs/$dracut_test-setup.log", 300;
+        assert_screen("dracut-root-block-success", 180);
```

Comment:
I would not use assert_screen, better grep the string from the log ?

Reply:
I would not run it in background and use `./test.sh --setup |& tee /tmp/logs/$dracut_test-setup.log`

Reply:
done